stl/inc/vector

```diff
@@ -578,9 +581,15 @@ private:
     }
 
 public:
-    vector(vector&& _Right) noexcept : _Mypair(_One_then_variadic_args_t{}, _STD move(_Right._Getal())) {
+    vector(vector&& _Right) noexcept
+        : _Mypair(_One_then_variadic_args_t{}, _STD move(_Right._Getal()), _Right._Mypair._Myval2._Myfirst,
+            _Right._Mypair._Myval2._Mylast, _Right._Mypair._Myval2._Myend) {
         _Mypair._Myval2._Alloc_proxy(_GET_PROXY_ALLOCATOR(_Alty, _Getal()));
-        _Move_construct(_Right, true_type{});
+        _Mypair._Myval2._Swap_proxy_and_iterators(_Right._Mypair._Myval2);
+
+        _Right._Mypair._Myval2._Myfirst = pointer();

```

Comment:
Should this be `nullptr` given that we always work with raw pointers for the members?

Reply:
`pointer()` is consistent with at least three pre-existing occurrences in this header. I wouldn't mind changing to `nullptr` - fancy pointers must meet the _`Cpp17NullablePointer`_ requirements which require `x = nullptr` to mean the same as `x = pointer()` - but if we do we should change all such occurrences (at least in this header) consistently.

Reply:
I pushed a global change for consistency - updating `<vector>`, `<xhash>`, and `<xstring>`, plus `unique_ptr` in `<memory>` which requires *Cpp17NullablePointer* (WG21-N4861 [unique.ptr.single.general]/3). The only remaining occurrence is `reset(pointer())` which needs to call that specific overload (`reset(nullptr)` would be an infinite loop; this could easily be refactored but it would be a larger change).

I hesitated because combining such a code cleanup with @Arzaghi's performance improvement is slightly more aggressive than we usually prefer to see in PRs (separate changes are easier to review, and make history easier to understand), but it'll be less work to just make the changes now. I take full responsibility if this messes anything up. :smile_cat:

Reply:
There's a failing test - working on a fix (looks like it revealed a deficiency in our test machinery which is good to find).